File: Leetcode/1695. Maximum Erasure Value/code.py
# Python Code: Sliding Window with Set
class Solution:
    def maximumUniqueSubarray(self, nums):
        ans = set()
        i = 0
        j = 0
        currSum = 0
        maxSum = 0

        while j < len(nums):
            # agr same nahi hai to insert kr do
            if nums[j] not in ans:
                ans.add(nums[j])
                currSum += nums[j]
                maxSum = max(currSum, maxSum)
                j += 1
            else:
                # agr same element hai to delete kr do
                ans.remove(nums[i])
                currSum -= nums[i]
                i += 1

        return maxSum

# Example usage
sol = Solution()
nums = [4, 2, 4, 5, 6]
print("Maximum score:", sol.maximumUniqueSubarray(nums))

File: Leetcode/1695. Maximum Erasure Value/code.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        unordered_set<int> ans;
        int i = 0, j = 0;
        int currSum = 0, maxSum = 0;

        while (j < nums.size()) {
            // agr same nahi hai to insert kr do 
            if (ans.find(nums[j]) == ans.end()) {
                ans.insert(nums[j]);
                currSum += nums[j];
                maxSum = max(currSum, maxSum);
                j++;
            } else {
                // agr same element hai to delete kr do
                ans.erase(nums[i]);
                currSum -= nums[i];
                i++;
            }
        }

        return maxSum;
    }
};

int main() {
    Solution sol;

    // Example input
    vector<int> nums = {4, 2, 4, 5, 6};

    // Output the result
    cout << "Maximum score: " << sol.maximumUniqueSubarray(nums) << endl;

    return 0;
}
